app/vmalert/remotewrite/remotewrite.go

```diff
@@ -164,44 +165,82 @@ func (c *Client) run(ctx context.Context) {
 				return
 			case <-ticker.C:
 				c.flush(ctx, wr)
-				wr = prompbmarshal.WriteRequest{}
-			case ts := <-c.input:
+			case ts, ok := <-c.input:
+				if !ok {
+					continue
```

Comment:
This will result in busy loop eating a CPU core after `c.input` is closed. Is this indendent?

Reply:
`c.input` may be closed only when `c.doneCh` is closed, so next switch will fall into shutdown handler. So I hope there will be no extra loops.